### ISSUE TITLE
sysrepo not creating necessary data files

### ISSUE THREAD

**Issue by plajjan** [sysrepo/sysrepo#384]
I'm asking sysrepo to install some YANG models and it seems to me that this should create some empty files in /etc/sysrepo/data that map up with the name of the model but for some reason it doens't create for all. Am I missing something?

```
root@ceaf428c010e:/# sysrepoctl --install --yang=/yang/ietf-inet-types.yang
root@ceaf428c010e:/# sysrepoctl --install --yang=/yang/lw4o6.yang
root@ceaf428c010e:/# ls -la /etc/sysrepo/data/
total 12
drwxr-xr-x 4 root root 4096 Oct  6 12:56 .
drwxr-xr-x 6 root root 4096 Oct  6 12:56 ..
drwxr-xr-x 2 root root 4096 Oct  6 12:56 internal
-rw-rw-rw- 1 root root    0 Oct  6 12:56 ts-lw4o6.candidate.lock
-rw-rw-rw- 1 root root    0 Oct  6 12:56 ts-lw4o6.persist
-rw-rw-rw- 1 root root    0 Oct  6 12:56 ts-lw4o6.running
-rw-rw-rw- 1 root root    0 Oct  6 12:56 ts-lw4o6.running.lock
-rw-rw-rw- 1 root root    0 Oct  6 12:56 ts-lw4o6.startup
-rw-rw-rw- 1 root root    0 Oct  6 12:56 ts-lw4o6.startup.lock
root@ceaf428c010e:/#
```

Why is only ts-lw4o6 created? Why not ietf-inet-types? This seems to later cause issues since the files doesn't exist:

```
[ERR] (sr_locking_set_lock_file_open:658) Insufficient permissions to lock the file '/etc/sysrepo/data/ietf-inet-types.running.lock'
```

**Comment by plajjan** [comment]
Okay so it actually tells me `Skipping installation of data files for module 'ietf-inet-types'...` and this might make sense since it's just data types.. nevertheless, I get the permission error so something needs to work differently ;)

**Comment by lukasmacko** [comment]
Hi,

yes, data files are not generated for yang models that contains only typedefs.

When the ERR log occured I guess, you were trying to lock datastore or were you performing some other operation? Did the operation failed or is that log just false alarm?

Regards,
Lukas

**Comment by plajjan** [comment]
Yes, I'm trying to lock the datastore.. and yes, you are right. This is just a log message.

NETCONF lock op succceeds:
```
<?xml version="1.0" encoding="UTF-8"?>
<rpc xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="1">
<lock><target><running/></target>
</lock>
</rpc>
]]>]]>
<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="1"><ok/></rpc-reply>]]>]]>
```

**Comment by lukasmacko** [comment]
Ok, I'll remove that confusing log.

**Comment by lukasmacko** [comment]
Fixed by #506
